File: bin_search/python/bin_search.py
l = [10, 21, 34, 42, 55, 68, 74, 83, 90, 106] # 10: 0 - 9

def int_bin_search(lst: list[int], target: int) -> int | None:
    low = 0
    high = len(lst) - 1 
    search_result = None

    while low <= high:
        mid = int((low + high) // 2) # get the index (round down the mid value)
        guess = lst[mid] # get the element 
        if guess == target:
            search_result = mid
            break
        if guess < target:
            low = mid + 1
        else:
            high = mid - 1
    
    return search_result

result = int_bin_search(l, 55)
print(result)

File: bin_search/javascript/bin_search.js
const l = [10, 21, 34, 42, 55, 68, 74, 83, 90, 106];

function bin_search(list, target) {
    let low = 0;
    let high = list.length - 1

    while (low <= high) {
        const mid = Math.floor((low + high) / 2)
        const current = list[mid]
        if (current == target) {
            return mid
        } 
        if (current < target) {
            low = mid + 1
        } else {
            high = mid - 1
        }
    }
    return null
}

console.log(bin_search(l, 90))